### DISCUSSION TITLE
Resources not loading on remote server

### DISCUSSION THREAD

**Discussion by cakiki** [mesa/mesa#1384] [Q&A] [answered]
Hi everyone! 

I'm very new to mesa and I'm still trying to find my way. 

I'm running an example from the docs on a online host. The server starts but the resources are being loaded from wrong addresses, so i'm getting a bunch of "Failed to load resource" for all CSS and JS files. You can see it yourself in the console: https://huggingface.co/spaces/cakiki/mesa 

The code I'm running is here: https://huggingface.co/spaces/cakiki/mesa/blob/main/app.py

**Comment by rht** [comment, marked as answer]
This is where the JS files can be accessed, e.g. https://hf.space/embed/cakiki/mesa/static/js/GridDraw.js. But huggingface.co assumes it is https://hf.space/static/js/GridDraw.js. If you want to make it work on huggingface.co, you have to fork Mesa, and in https://github.com/projectmesa/mesa/blob/main/mesa/visualization/templates/modular_template.html, change the `/static/...` with absolute path `https://hf.space/embed/cakiki/mesa/static/...`.

**Reply by rht** [reply]
Actually, before doing that, first try changing "/static/..." to "static/..." and see if this works.

**Reply by cakiki** [reply]
changing to "static/" worked! Thank you so much.

**Reply by rht** [reply]
Good to know! I think we should remove those slashes in Mesa so that it works out of the box in any hosting service. This is fine since we only have a single HTML page to serve, and won't go to more subdirectories relative to the Tornado server's path. Any thoughts @Corvince ?

**Reply by cakiki** [reply]
Sorry for the follow-up question, but I am now facing a different issue. The websocket connection of my mesa app is trying to access the root domain for the hosting website (`wss://hf.space/ws`); wheras I'd need it to connect to `wss://hf.space/embed/cakiki/mesa/ws`

Will it be enough to hardcode that here: https://github.com/projectmesa/mesa/blob/main/mesa/visualization/templates/js/runcontrol.js#L132-L136

**Reply by rht** [reply]
Yes, that should do.
